utils/download.ts

```diff
@@ -0,0 +1,29 @@
+import { useAuth0 } from "@auth0/auth0-vue";
+import type { TrackModel } from "@bcc-code/bmm-sdk-fetch";
+
+function downloadTrack(track: TrackModel, accessToken: string) {
+  const file = defaultFileForTrack(track);
+  const url = file?.url;
+  if (file && url) {
+    const authUrl = authorizedUrl(url, accessToken);
+    window.location.href = `${authUrl}&download=true`;
+  }
+}
+
+export type DownloadResult = "no-permission" | "downloaded";
+export function useTrackDownloader() {
```

Comment:
This is a composable, and should be in the `composables` folder, strictly speaking. 
It is quite practical to keep everything in one place, but I think we should try following Nuxt conventions.